File: app/src/main/java/com/davidbronn/fragmentrevision/broadcast_communicate/BroadCastFragment.java
package com.davidbronn.fragmentrevision.broadcast_communicate;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.LocalBroadcastManager;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.EditorInfo;
import android.widget.TextView;

import com.davidbronn.fragmentrevision.R;
import com.davidbronn.fragmentrevision.databinding.FragmentBroadCastBinding;

public class BroadCastFragment extends Fragment {

    private FragmentBroadCastBinding binding;

    public static BroadCastFragment newInstance(Bundle bundle) {
        BroadCastFragment fragment = new BroadCastFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_broad_cast, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.edQueryText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
            @Override
            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
                if (actionId == EditorInfo.IME_ACTION_DONE) {
                    Intent intent = new Intent(BroadcastActivity.BROADCAST);
                    intent.putExtra("message", binding.edQueryText.getText().toString());
                    LocalBroadcastManager.getInstance(getActivity()).sendBroadcast(intent);
                }
                return false;
            }
        });
    }
}

File: app/src/main/java/com/davidbronn/fragmentrevision/usingBroadCast/BroadCastFragment.kt
package com.davidbronn.fragmentrevision.usingBroadCast

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.widget.addTextChangedListener
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import com.davidbronn.fragmentrevision.R
import com.davidbronn.fragmentrevision.databinding.FragmentBroadCastBinding

class BroadCastFragment : Fragment() {

    private var binding: FragmentBroadCastBinding? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_broad_cast, container, false)
        return binding?.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding?.edQueryText?.addTextChangedListener {
            broadCastMessage(it.toString())
        }
    }

    private fun broadCastMessage(message: String) {
        val intent = Intent(MainActivity.BROADCAST)
        intent.putExtra("message", message)
        LocalBroadcastManager.getInstance(activity!!).sendBroadcast(intent)
    }
}
